app/lib/task/scheduler.dart

```diff
@@ -67,7 +65,10 @@ Future<void> schedule(
 
       // If terminated or older than maxInstanceAge, delete the instance...
       final isTerminated = instance.state == InstanceState.terminated;
-      final isTooOld = instance.created.isBefore(clock.agoBy(_maxInstanceAge));
+      final maxInstanceAge = Duration(
+        hours: activeConfiguration.maxTaskRunHours,
+      );
+      final isTooOld = instance.created.isBefore(clock.agoBy(maxInstanceAge));
```

Comment:
Nit: This may be easier to read if we had the time going forward, e.g. `instance.created.add(maxInstanceAge).isBefore(clock.now())`